Source-Code/FlareDraw.cs

```diff
@@ -192,9 +192,22 @@ private void GenerateBodyFlares()
             //ListChildren(sm.systemPrefab.rootBody, 0);
             //--- HACK--
 
+            //If Kerbin is parented to the Sun, set its SMA- otherwise iterate
+            //through celestial bodies to locate which is parented to the Sun
+            //and has Kerbin as a child. Set the highest parents SMA to kerbinSMA
             if(BodyFlare.kerbinSMA <= 0.0)
             {
-                BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                if(FlightGlobals.Bodies[1].referenceBody == FlightGlobals.Bodies[0])
+                    BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                else
+                {                    
+                    foreach(CelestialBody current in FlightGlobals.Bodies)
```

Comment:
Okay.  Not a problem.

>I actually wanted to use a for() statement that broke out when it was
located but wasn't sure how to find how many bodies were in the index for
the range.

You mean `FlightGlobals.Bodies.Count`?